Index: src/main/java/pl/edu/wit/studentManagement/translations/Translator.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package pl.edu.wit.studentManagement.translations;\r\n\r\nimport java.util.ResourceBundle;\r\n\r\npublic final class Translator {\r\n    private static Language language = Language.ENGLISH;\r\n    private static ResourceBundle bundle = loadBundle();\r\n\r\n    private Translator() {}\r\n\r\n    private static ResourceBundle loadBundle() {\r\n        return ResourceBundle.getBundle(\"messages\",language.getLocale());\r\n    }\r\n\r\n    public static void setLanguage(Language lang) {\r\n        language = lang;\r\n        bundle = loadBundle();\r\n    }\r\n\r\n    public static String translate(String key) {\r\n        return bundle.getString(key);\r\n    }\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/pl/edu/wit/studentManagement/translations/Translator.java b/src/main/java/pl/edu/wit/studentManagement/translations/Translator.java
--- a/src/main/java/pl/edu/wit/studentManagement/translations/Translator.java	(revision 56705e58a32afcd3df647248fe420aff7a61cbf1)
+++ b/src/main/java/pl/edu/wit/studentManagement/translations/Translator.java	(date 1750587690233)
@@ -12,6 +12,10 @@
         return ResourceBundle.getBundle("messages",language.getLocale());
     }
 
+    public static Language getLanguage() {
+        return language;
+    }
+
     public static void setLanguage(Language lang) {
         language = lang;
         bundle = loadBundle();
